backend/src/main/java/de/neuefische/capstone/userplants/UserPlantService.java

```diff
@@ -0,0 +1,32 @@
+package de.neuefische.capstone.userplants;
+
+import de.neuefische.capstone.plantcomponents.Plant;
+import de.neuefische.capstone.plantcomponents.PlantRepo;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.security.Principal;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class UserPlantService {
```

Comment:
Instead of throwing exceptions, I would return `Optional<Plant>` in the first three methods. If you do that, you could use `ResponseEntity.of` in the controller.